Simplified/MyBooks/NYPLMyBooksDownloadCenter.m

```diff
@@ -558,14 +558,34 @@ - (void)deleteLocalContentForBookIdentifier:(NSString *const)identifier account:
       }
 #endif
 
+#if defined(LCP)
+      if ([LCPAudiobooks canOpenBook:book]) {
+        LCPAudiobooks *lcpAudiobooks = [[LCPAudiobooks alloc] initFor:bookURL];
+        dict = [[lcpAudiobooks contentDictionary] mutableCopy];
+        dict[@"id"] = book.identifier;
+      }
+#endif
       [[AudiobookFactory audiobook:dict ?: json] deleteLocalContent];
+      
+#if defined(LCP)
+      if ([LCPAudiobooks canOpenBook:book]) {
+        if ([[NSFileManager defaultManager] fileExistsAtPath:bookURL.path]) {
+          NSError *error = nil;
+          [[NSFileManager defaultManager] removeItemAtURL:bookURL error:&error];
+          if (error) {
+            [NYPLErrorLogger logError:error
+                              summary:@"Failed to delete LCP audiobook local content"
+                              message:NULL
+                             metadata:@{ @"book": [book loggableShortString] }];
+          }
+        }
+      }
+#endif
```

Comment:
do you think it's possible to move all the code between 544-583 into a separate method? Just trying to avoid expanding this gigantic switch case and limit the pain  :)

Reply:
looks more like a necessity than a possibility, moved into a separate method.